### PR TITLE
Handle resources found in the child JARs of a WAR as well as the contents of the war

### LINKED ISSUE #39: ServiceLoader not loading from all JARs.
@bartoszmajsak  I'm not sure if we're still using github issues or JIRA, feel free to redirect!

##### Issue Overview

The classloader around service loaders doesn't pick up all classes when using TCCL.

##### Expected Behaviour

Given I have a ShrinkWrap archive that creates a WAR, and two JAR files (jarA and jarB), when inside jarB I use `ServiceLoader.load(SomeClass, TCCL)` I get back all services defined in the WAR, jarA and jarB.

##### Current Behaviour

The contents of jarA are ignored.

##### Steps To Reproduce
If you check out https://svn.apache.org/repos/asf/geronimo/components/config/trunk/ and run `mvn clean install -PWeld3` you'll see that several tests fail due to missing config properties.
The config properties are not loaded as the service loader isn't finding the relevant services.  Debug `DefaultConfigBuilder.build` to see the service loader's behavior.

### PR DESCRIPTION
**Pull request by johnament**
#### Short description of what this resolves:

This closes #39 
Specifically, when a resource is found in a JAR, also consider that when doing a find.

#### Changes proposed in this pull request:

- New test that verifies this behavior.
- Recurse through the child JARs to find URLs.

**Fixes**: #39

### CODE REVIEW

File: impl/src/test/java/org/jboss/arquillian/container/weld/embedded/ExtensionInJARTestCase.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                .addAsServiceProvider(Extension.class, WasCalledExtension.class)
+                .addClasses(WasCalledExtension.class)
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+        return ShrinkWrap.create(WebArchive.class)
+                .addAsLibraries(jar)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    @Test
+    public void shouldBeAbleToInjectBeanAsInstanceVariable() throws Exception {
```
Comment (tremes) [inline impl/src/test/java/org/jboss/arquillian/container/weld/embedded/ExtensionInJARTestCase.java]: Test method name should be changed.

File: impl/src/test/java/org/jboss/arquillian/container/weld/embedded/ExtensionInJARTestCase.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.enterprise.inject.spi.Extension;
+
+@RunWith(Arquillian.class)
+public class ExtensionInJARTestCase {
```
Comment (tremes) [inline impl/src/test/java/org/jboss/arquillian/container/weld/embedded/ExtensionInJARTestCase.java]: The name of the test class doesn't fully reflect the test case IMHO. I would choose some other name like `ExtensionInLibJarTestCase` or maybe `WebArchiveWithLibJarExtensionTestCase` or something like that.

File: impl/src/test/java/org/jboss/arquillian/container/weld/embedded/ExtensionInJARTestCase.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Deployment
+    public static Archive<?> createdeployment() {
+        JavaArchive jar = ShrinkWrap.create(JavaArchive.class)
+                .addAsServiceProvider(Extension.class, WasCalledExtension.class)
+                .addClasses(WasCalledExtension.class)
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+        return ShrinkWrap.create(WebArchive.class)
+                .addAsLibraries(jar)
```
Comment (tremes) [inline impl/src/test/java/org/jboss/arquillian/container/weld/embedded/ExtensionInJARTestCase.java]: What about to add another lib jar to the test? I think you described it in the issue summary that there are two jars.

Comment (johnament) [inline impl/src/test/java/org/jboss/arquillian/container/weld/embedded/ExtensionInJARTestCase.java]: Turns out it has nothing to do with the number of JARs, none were being loaded.

File: impl/src/test/java/org/jboss/arquillian/container/weld/embedded/ExtensionInJARTestCase.java (hunk 1 above)
Comment (johnament) [inline impl/src/test/java/org/jboss/arquillian/container/weld/embedded/ExtensionInJARTestCase.java]: What would you like the method to be named?

Comment (tremes) [inline impl/src/test/java/org/jboss/arquillian/container/weld/embedded/ExtensionInJARTestCase.java]: I don't know `libExtensionLoaded` or `lifecycleEventObserved`. Depends also on the test class name. I think it should be clear there's war with lib/jar with extension.

**Comment by johnament** [comment]
Let me know if you're good with these.

**Comment by tremes** [comment]
Thank's it's fine from my point of view. I'll wait whether other reviewers will respond.

**Comment by johnament** [comment]
I would appreciate if a release could be cut after these changes are merged.

**Comment by tremes** [comment]
Merged. Thanks

**Pull request merged**